### PR TITLE
Use edx standard dependency management and fix Makefile

### PR DESCRIPTION
**Pull request by timmc-edx**
- Use requirements file when advertising dependencies in setup.py
- Fix PHONY targets in Makefile (was interfering with `make requirements`)
- Use pip-sync when installing requirements
- Rename requirements.in to base.in (and corresponding txt file)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -13,7 +13,7 @@ ENV LANGUAGE en_US:en
 ENV LC_ALL en_US.UTF-8
 
 WORKDIR /app
-ADD requirements/requirements.txt /app/requirements/requirements.txt
+ADD requirements/base.txt /app/requirements/base.txt
```
Comment (timmc-edx) [inline Dockerfile:16]: I think this is pointless because of the later `ADD . /app` but hey, consistency.

Comment (coryleeio) [reply Dockerfile:16]: Adding it here means we wont reinstall dependencies on subsequent docker builds unless the requirements have actually changed.

Comment (coryleeio) [reply Dockerfile:16]: It is useful when doing dev for a lot of apps

Comment (timmc-edx) [reply Dockerfile:16]: Ah! Makes sense. :-)

**Review by coryleeio — APPROVED** [review]

**Pull request merged**